### DISCUSSION TITLE
Respond based on value in previous query

### DISCUSSION THREAD

**Discussion by JohnnySheppard** [mockoon/mockoon#601] [Help] [answered]
Hi,

First off, I'm a bit new to all this, but I've come across Mockoon - and it looks awesome! I'm working on a project to automate some PTZ cameras that have a REST API. I won't have much access to the cameras much, so this tool will be absolutely perfect for testing my system.

So the question I have is, can I make a request (that would, for example, turn the cameras on/off) to Mockoon, then run another request to check the status of the power. This request would return its value based on what had been set in the previous request?

I guess what I'm asking is, is there a way to set a variable with a request, then read that variable from a second request?

It's no problem if not, I have thought of a workaround, but it would be amazing if it did support this kind of functionality.

Some examples to try to explain what I'm trying to do:

```text
# Turn the camera on
http://camera1.lan/cgi-bin/aw_ptz?cmd=%23O1&res=1

# Turn the camera off
http://camera1.lan/cgi-bin/aw_ptz?cmd=%23O0&res=1
```

And then based on the value that has been set by either of the requests:

```text
http://camera1.lan/cgi-bin/aw_ptz?cmd=%23O&res=1
# Would return p1/p0 depending on the value set by the previous requests
```

Regardless of the answer, thanks again for this amazing software.

Johnny :)

**Comment by 255kb** [comment, marked as answer]
Hi @JohnnySheppard We are glad you like our tool so far 😊
There is currently no relation between calls. There is one issue that could solve your problem but it's not planned yet: https://github.com/mockoon/mockoon/issues/158
As a workaround, you could use the [rules system](https://mockoon.com/docs/latest/route-responses/dynamic-rules/) and serve different responses depending on the `cmd` query param value: 

![image](https://user-images.githubusercontent.com/7489814/143580338-801673c6-eb16-4d28-9a12-4e34f51b35e4.png)

If you want to simulate some kind of on/off switch, you could create 2 responses with the expected on/off body response and use the "sequential response" option:
![image](https://user-images.githubusercontent.com/7489814/143580589-958888b8-5e41-41e1-9c42-28a88ab69330.png)

Let me know if this works for your use case!

**Comment by JohnnySheppard** [comment]
Hi @255kb,

Thanks very much for the response. Yes your workaround would work and I may use that, I have another idea too which is to duplicate the whole endpoint, but change the response, then have one endpoint as the "Camera's On" endpoint, and the other that's the "Camera's Off" endpoint. Then toggle whichever endpoint I want at the time. These two workarounds will do for now. 

The issue you linked looks like it would do exactly what I want though, so I'll keep an eye open for that in case it finds its way onto the roadmap. No pressure though :)
